### PR TITLE
Fix [#187355106] profile settings

### PR DESCRIPTION
**Pull request by pacifiquemboni**
1.What does this PR do?
Profile Page Settings

2.Description
Once user user is done with registration he/she will be able to see his/her profile(includes personal information and billing address) and he/she will be able to update it by adding additional information such as gender, birthdate etc.

3.Acceptance Criteria
I want to be able to Edit my profile, but i should NOT be able to edit another users profile
relevant pivotal tracker story https://www.pivotaltracker.com/story/show/187355106 
swagger doc
Screenshot 2024-05-25 105729
when you try to get single user and update user profile without being authonticated
Screenshot 2024-05-25 114614

4.steps followed to test this feature
i)clone develop branch git clone https://github.com/atlp-rwanda/e-commerce-ones-and-zeroes-bn.git
ii)Checkout to the branch using git checkout ft-profile-page-setting
iii)setup environment variables (.env file) following .env.example file
PORT="your port number"
D
ATABASE_URL="your_database_url"
NODE_ENV="your_database_level"(it shoulbe development, test or production)
iv)run npm run build to convert .ts files into .js format
v) run npm migrate:all to migrate provided models in you database
vi)run npm run dev to start the server

5.use :
npm run test to run tests and npx jest --coverage to see test coverage
http://localhost:your_port_number/docs to see swagger documenta

### CODE REVIEW

**Review by lilika67 — APPROVED** [review]

**Review by Celestin4 — COMMENTED** [review]
These functionalities related to profile setting requiere user to be authenticated. I dont see where we need token to perform action.

**Review by Celestin4 — APPROVED** [review]

File: src/routes/index.ts [hunk 1]
```diff
@@ -10,6 +11,8 @@ router.get('/', (req: Request, res: Response) => {
 
 // Use userRoute for paths starting with '/users'
 router.use('/users', userRoute);
+router.use('/user', userRoute);
+router.use('/user', updateUserRoutes);
```
Comment (UmuhireJessie) [inline src/routes/index.ts]: All routes of user entity should have the same prefix either `/user` or `/users`

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Check my comments and add screenshot of how profile setting feature works

File: src/docs/Users/updateUser.ts [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+                  message: { type: 'string' },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+  '/api/users/editUser/{id}': {
```
Comment (UmuhireJessie) [inline src/docs/Users/updateUser.ts]: Avoid using words like 'editUser' or 'update-user' in routes

File: src/routes/updateUserRoutes.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  authMiddleware.checkRole,
+  UserController.getUsers,
+);
+router.get(
+  '/:id',
+  authMiddleware.isAuthenticated,
+  UserController.getSingleUser,
+);
+router.patch(
+  '/editUser/:id',
```
Comment (UmuhireJessie) [inline src/routes/updateUserRoutes.ts]: Same here!

File: src/docs/Users/updateUser.ts
Comment (UmuhireJessie) [inline src/docs/Users/updateUser.ts:1]: Can we have one file for user routes?

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Good job @pacifiquemboni, see my comments above

**Review by lilika67 — APPROVED** [review]

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**